Mathlib/MeasureTheory/Measure/Lebesgue/EqHaar.lean

```diff
@@ -79,6 +83,42 @@ theorem Basis.parallelepiped_basisFun (ι : Type*) [Fintype ι] :
     · exact zero_le_one
 #align basis.parallelepiped_basis_fun Basis.parallelepiped_basisFun
 
+ /-- A parallelepiped can be expressed on the standard basis. -/
+ theorem Basis.parallelepiped_eq_map  {ι E : Type*} [Fintype ι] [NormedAddCommGroup E]
+    [NormedSpace ℝ E] (b : Basis ι ℝ E) :
+    b.parallelepiped = (TopologicalSpace.PositiveCompacts.piIcc01 ι).map b.equivFun.symm
+    b.equivFunL.symm.continuous
+      (by have := FiniteDimensional.of_fintype_basis b
+          exact LinearMap.isOpenMap_of_finiteDimensional _ b.equivFun.symm.surjective) := by
+  rw [← Basis.parallelepiped_basisFun]
+  refine (congrArg _ ?_).trans (Basis.parallelepiped_map _ _)
+  classical
+  ext
+  simp only [map_apply, Pi.basisFun_apply, equivFun_symm_apply, LinearMap.stdBasis_apply',
+    Finset.sum_univ_ite]
+
+theorem Basis.parallelepiped_measurable {ι E : Type*} [Fintype ι] [NormedAddCommGroup E]
+    [NormedSpace ℝ E] [MeasurableSpace E] [OpensMeasurableSpace E] (b : Basis ι ℝ E) :
+    MeasurableSet (b.parallelepiped : Set E) := by
+  rw [Basis.parallelepiped_eq_map, PositiveCompacts.coe_map, LinearEquiv.image_symm_eq_preimage]
+  exact (continuous_equivFun_basis _).measurable (PositiveCompacts.piIcc01_measurable ι)
```

Comment:
Again, could you not use [IsCompact.measurableSet](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.html#IsCompact.measurableSet)? Maybe it's worth adding `PositiveCompacts.measurableSet`.

Reply:
I have removed this result and used [IsCompact.measurableSet](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.html#IsCompact.measurableSet) instead. Although, it might be better to leave the result with the one-line proof using [IsCompact.measurableSet](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.html#IsCompact.measurableSet).